runner/utils/driver/docker/docker.go

```diff
@@ -0,0 +1,255 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/sirupsen/logrus"
+	"github.com/txn2/txeh"
+
+	"github.com/spaceuptech/space-cloud/runner/model"
+	"github.com/spaceuptech/space-cloud/runner/utils/auth"
+)
+
+type docker struct {
+	client *client.Client
+	auth   *auth.Module
+}
+
+func NewDockerDriver(auth *auth.Module) (*docker, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		logrus.Errorf("error creating docker module instance in docker in docker unable to initialize docker client - %v", err)
+		return nil, err
+	}
+	return &docker{client: cli, auth: auth}, nil
+}
+
+// ApplyService creates containers for specified service
+func (d *docker) ApplyService(ctx context.Context, service *model.Service) error {
+	// remove containers if already exits
+	if err := d.DeleteService(ctx, service.ID, service.ProjectID, service.Version); err != nil {
+		logrus.Errorf("error applying service in docker unable delete existing containers - %v", err)
+		return err
+	}
+
+	// todo check host file overiding concern
+	// client for CRUD operation on host file
+	// default location /etc/hosts
+	hostFile, err := txeh.NewHostsDefault()
```

Comment:
Where is this creating a hosts file? We don't want this to over write the users `/etc/hosts` file